test/request_source/request_source_plugin_test.cc

```diff
@@ -0,0 +1,114 @@
+#include "envoy/common/exception.h"
+
+#include "external/envoy/source/common/config/utility.h"
+#include "external/envoy/test/mocks/stats/mocks.h"
+#include "external/envoy/test/test_common/file_system_for_test.h"
+#include "external/envoy/test/test_common/utility.h"
+
+#include "common/request_source_plugin_impl.h"
+
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+namespace {
+using nighthawk::request_source::DummyPluginRequestSourceConfig;
+using nighthawk::request_source::FileBasedPluginRequestSourceConfig;
+using ::testing::Test;
+
+class DummyRequestSourcePluginTest : public Test {
+public:
+  DummyRequestSourcePluginTest() : api_(Envoy::Api::createApiForTest(stats_store_)) {}
+  Envoy::Api::ApiPtr api_;
+  Envoy::Stats::MockIsolatedStatsStore stats_store_;
+}; // RequestSourcePluginTest
+class FileBasedRequestSourcePluginTest : public Test {
+public:
+  FileBasedRequestSourcePluginTest() : api_(Envoy::Api::createApiForTest(stats_store_)) {}
+  Envoy::Api::ApiPtr api_;
+  Envoy::Stats::MockIsolatedStatsStore stats_store_;
+  nighthawk::request_source::FileBasedPluginRequestSourceConfig
+  MakeFileBasedPluginConfigWithTestYaml(const std::string& request_file) {
+    nighthawk::request_source::FileBasedPluginRequestSourceConfig config;
+    config.mutable_uri()->assign("http://foo/");
+    config.mutable_file_path()->assign(request_file);
+    return config;
+  }
+}; // RequestSourcePluginTest
+
+TEST_F(DummyRequestSourcePluginTest, CreateEmptyConfigProtoCreatesCorrectType) {
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<RequestSourcePluginConfigFactory>(
+          "nighthawk.dummy-request-source-plugin");
+  const Envoy::ProtobufTypes::MessagePtr empty_config = config_factory.createEmptyConfigProto();
+  const nighthawk::request_source::DummyPluginRequestSourceConfig expected_config;
+  EXPECT_EQ(empty_config->DebugString(), expected_config.DebugString());
+  EXPECT_TRUE(Envoy::MessageUtil()(*empty_config, expected_config));
+}
+TEST_F(DummyRequestSourcePluginTest, FactoryRegistrationUsesCorrectPluginName) {
```

Comment:
nit: missing newline here and below in a couple of other spots.

Reply:
Cleaned those up. Thanks.